### ISSUE TITLE
ECR image tags should have different value for each environments.

### ISSUE THREAD

**Issue by mather** [aws/copilot-cli#2927]
Docker image tags for each environments are currently the same value. (shrinked `$CODEBUILD_BUILD_ID`)
https://github.com/aws/copilot-cli/blob/fdf78af706272c2113ba593db7e7bd983dba2622/internal/pkg/template/templates/cicd/buildspec.yml#L38-L47
Even if you overwrite some variables in `environments` of `manifest.yml`, it will be updated with the value of the last `environment` processed.
So, `tag` value should be specified for each `environment` and `$CODEBUILD_BUILD_ID`. (While taking into account the character limit of image tag)

I can fix it for my `buildspec.yml`, but the template should be fixed.

**Comment by iamhopaul123** [comment]
Hello @mather. It seems like for the `$tag` we always use `$CODEBUILD_BUILD_ID` (for example, `codebuild-demo-project:b1e6661e-e4f2-4156-9ab9-82a19EXAMPLE`) which seems to be independent from environments. Did you override this env var in your manifest so that you want different tags for each environment?

**Comment by mather** [comment]
> Did you override this env var in your manifest so that you want different tags for each environment?

Yes, I overrode `<env>.image.build.args` to change ARG variable in Dockerfile.
Here is an example:
```
image:
  build:
    args:
      FOO: bar
environments:
  test:
    image:
      build:
        args:
          FOO: baz
```

In default `buildspec.yml`, it runs `docker build` once for single `tag` value.
This is a different behavior from manual deployment:

```
$ copilot svc deploy -n app -e test
$ copilot svc deploy -n app -e production
```

I thought it should behave the same as a manual deployment.

**Comment by iamhopaul123** [comment]
Gotcha! That makes a lot more sense to me. Just to double check you just want Copilot to concatenate the env name with `CODEBUILD_BUILD_ID` to make sure each deploy uses different image? Otherwise they'll use only the lastly built one?

**Comment by mather** [comment]
You can rebuild docker image and create tag for each environments.
If each build will generate the same image, build steps will be skipped by using cache, and push single image with multiple tags.

**Comment by efekarakus** [comment]
This is now released 🎉 in v1.13.0 https://github.com/aws/copilot-cli/releases/tag/v1.13.0 !
